docs/tests-ui.rst

```diff
@@ -1,177 +1,148 @@
-WebDriver API which lets us write automated testing via JavaScript. Intern is
-an open source project created and maintained by SitePen_.
+===================
+Client-side Testing
+===================
 
-In 2016, we started to convert our client-side tests to Python, to standardize
-on common Mozilla QA processes, and as part of an effort to add client-side
-tests to the development pipline.  The Intern tests described here are no
-longer run for development and deployment, and have not been updated for the
-Docker development environment.
+Kuma has a suite of functional tests using `Selenium`_ and `pytest`_. This allows us
+to emulate users interacting with a real browser. All these test suites live in
+the /tests/ directory.
 
-.. _SitePen: http://sitepen.com
-.. _Intern: https://theintern.github.io/
+The tests directory comprises of:
 
-Installing Dependencies
------------------------
+   * ``/functional`` contains pytest tests.
+   * ``/pages`` contains Python pypom page objects.
+   * ``/utils`` contains helper functions.
 
-1. Go to the ``tests/ui/`` directory::
+.. _`Selenium`: http://docs.seleniumhq.org/
+.. _`pytest`: http://pytest.org/latest/
 
-    cd tests/ui
+Setting up test virtual environment
+===================================
 
-2. Use ``npm`` to install Intern::
+#. In the terminal go to the directory where you have downloaded kuma
 
-    npm install intern@^3.0
+#. Create a virtual environment for the tests to run in (the example uses the
+   name mdntests)::
 
-.. warning:: Do *not* install Intern globally -- path issues may occur.
+   $ virtualenv mdntests
 
-Running Tests
--------------
+#. Activate the virtual environment::
 
-On your machine
-~~~~~~~~~~~~~~~
+   $ source mdntests/bin/activate
 
-1. Install `JDK <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`_
+#. Install the requirements the tests need to run::
 
-2. Download the most current `release of Selenium <http://selenium-release.storage.googleapis.com/index.html>`_ standalone server. (It's the ``.jar`` file.)
+   $ pip install -r requirements/test.txt
 
-.. note:: Firefox should work out of the box. You need to install `Chrome <https://sites.google.com/a/chromium.org/chromedriver/>`_ and `Safari <https://code.google.com/p/selenium/wiki/SafariDriver>`_ drivers yourself.
+You may want to add your virtual environment folder to your local .gitignore
+file
 
-3. From the command line, start WebDriver::
+Running the tests
+=================
 
-    # Substitute your WebDriver version in the `#` chars
-    java -jar /path/to/selenium-server-standalone-#.#.#.jar
+The minimum amount of information necessary to run the tests against the staging
+server is the directory the tests are in and which driver to use.
 
-4. Go to the ``tests/ui/`` directory::
+In the virtual environment run::
 
-    cd tests/ui
+   $ py.test tests/functional/ --driver Firefox --driver-path /path/to/geckodriver
```

Comment:
This is the first time the driver is mentioned. It may be necessary to point people at documentation for downloading the ``geckodriver`` first. It is not obvious from an internet search how to get it, and it is also not obvious how [these documents](https://developer.mozilla.org/en-US/docs/Mozilla/QA/Marionette/WebDriver) apply.

After downloading ``geckodriver`` for OS X from the [release page](https://github.com/mozilla/geckodriver/releases), I get an error with this command:

```
py.test tests/functional/ --driver Firefox --driver-path ~/Downloads/geckodriver

======================================================= test session starts ========================================================
platform darwin -- Python 2.7.12, pytest-3.0.3, py-1.4.31, pluggy-0.4.0
driver: Firefox
sensitiveurl: .*
rootdir: /Users/john/src/kuma, inifile: pytest.ini
plugins: base-url-1.1.0, html-1.10.1, rerunfailures-1.0.2, selenium-1.4.0, variables-1.4, xdist-1.15.0
collected 29 items / 1 errors 

============================================================== ERRORS ==============================================================
________________________________________ ERROR collecting tests/functional/test_notfound.py ________________________________________
ImportError while importing test module '/Users/john/src/kuma/tests/functional/test_notfound.py'.
Original error message:
'No module named urls'
Make sure your test modules/packages have valid Python names.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
===================================================== 1 error in 0.12 seconds ======================================================

```
This is running against the master branch, not this branch.